### PR TITLE
Prevent duplicate incidents

### PR DESCRIPTION
**Pull request by DBZclaroty**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [ ] Ready
- [x] In Hold - (branch became unsynced. openned a new PR: https://github.com/demisto/content/pull/32765)

## Related Issues
None

## Description
When fetching alerts and assets, only compare timestamp using "gt" instead of "gte".
This prevents duplicates.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/Claroty/Integrations/Claroty/Claroty.py [hunk 1]
```diff
@@ -601,22 +601,23 @@ def fetch_incidents(client: Client, last_run, first_fetch_time):
...
-            'name': item.get('Description', None),
-            'occurred': incident_created_time.strftime(DATE_FORMAT),
-            'severity': CTD_TO_DEMISTO_SEVERITY.get(item.get('Severity', None), None),
-            'rawJSON': json.dumps(item)
-        }
-
-        incidents.append(incident)
-        current_rids.append(item["ResourceID"])
+        rid = item["ResourceID"]
+        if rid not in last_run_rids:
```
Comment (YuvHayun) [inline Packs/Claroty/Integrations/Claroty/Claroty.py:606]: so if I understand correctly, the change is that now you check if the rid was a part of the previous call?

File: Packs/Claroty/Integrations/Claroty/Claroty_test.py [hunk 2]
```diff
@@ -137,16 +138,36 @@ def test_claroty_authentication(mocker, requests_mock):
...
-    nextcheck, incidents = fetch_incidents(client, {'lastRun': dateparser.parse("2018-10-24T14:13:20+00:00")}, first_fetch_time)
+    next_check, incidents = fetch_incidents(client, {'lastRun': dateparser.parse("2018-10-24T14:13:20+00:00")}, first_fetch_time)
 
-    assert nextcheck['last_fetch']
+    assert next_check['last_fetch']
     assert isinstance(incidents, list)
     assert incidents[0]['severity'] == 4  # Demisto severity is higher by one (doesn't start at 0)
     assert isinstance(incidents[0]['name'], str)
 
+    rids_from_incidents = get_rids_from_incidents(incidents)
```
Comment (YuvHayun) [inline Packs/Claroty/Integrations/Claroty/Claroty_test.py:157]: looks good, please add a given-when-then doc-strings to the test here.

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Looks good in general, minor comments.
Also, please note you need to update release notes (can be done by running `demisto-sdk update-release-notes -g`)

File: Packs/Claroty/Integrations/Claroty/Claroty.py (hunk 1 above)
Comment (DBZclaroty) [reply Packs/Claroty/Integrations/Claroty/Claroty.py:606]: Exactly.
If this is unclear, I can use a more intuitive implementation, for example:
```python
new_items = [item for item in items if item["ResourceID"] not in last_run_rids]

for item in new_items:
    # Make datetime object unaware of timezone for comparison
    parsed_date = dateparser.parse(item['Timestamp'])
    assert parsed_date is not None, f"failed parsing {item['Timestamp']}"
    incident_created_time = parsed_date.replace(tzinfo=None)

    incident = {
        'name': item.get('Description', None),
        'occurred': incident_created_time.strftime(DATE_FORMAT),
        'severity': CTD_TO_DEMISTO_SEVERITY.get(item.get('Severity', None), None),
        'rawJSON': json.dumps(item)
    }

    incidents.append(incident)
    current_rids.append(rid)
```

**Review by DBZclaroty — COMMENTED** [review]
Thanks for the review! Let me know if this change would make it more clear.

**Comment by YuvHayun** [comment]
Seems like your branch is unsynced..
I'd suggest trying to sync it:
1. Save all of your changes locally or in a new branch.
2. Enter your forked repository page in GitHub.
3. At the top of the page, you will probably see something like this, please ensure you are in the master branch.
image

5. Click on "Sync fork" button on the top right side of the page.
image

6. The following message will appear (This branch has conflicts that must be resolved).

7. Click on the button "Discard commits". Note that for this action to work, you should have permission to do a force push in the forked repo. If you don't have one, ask the forked repo Owner to do that.
8. Enter your branch and repeat the same steps (4-6).
9. Check your PR, and see if your changes have been saved. If yes - you finished the update process.
10. If your PR does not show your changes, open a new PR with the changes that you saved at the beginning of the process.
11. In case you opened a new PR please close the old one, and update your reviewer with the new PR link.
12. Please paste on the new open PR the URL to your previous PR so our team member can continue the review from the same place.

File: Packs/Claroty/ReleaseNotes/1_0_27.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+#### Integrations
+
+##### Claroty
+
+- Improve test of 'fetch_incidents', to also test for the same incidents being
+  retrieved multiple times.
+- When fetching incidents, only return incidents that were not received previously.
+- Improve type-hints in 'Claroty.py'.
```
Comment (YuvHayun) [inline Packs/Claroty/ReleaseNotes/1_0_27.md:9]:
```suggestion
- Updated the Docker image to: *demisto/python3:3.10.13.86272*.
- Fixed an issue where **fetch-incidents** returned duplicated incidents.
```

File: Packs/Claroty/Integrations/Claroty/Claroty_test.py [hunk 4]
```diff
@@ -137,16 +138,36 @@ def test_claroty_authentication(mocker, requests_mock):
 
 
 def test_claroty_fetch_incidents(mocker, requests_mock):
+    def get_rids_from_incidents(incidents: list[dict[str, str]]) -> set[int]:
```
Comment (YuvHayun) [inline Packs/Claroty/Integrations/Claroty/Claroty_test.py:141]: given-when-then doc strings are still missing.

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Minor comments.

**Comment by YuvHayun** [comment]
moved [here](https://github.com/demisto/content/pull/32765)

**Comment by DBZclaroty** [comment]
@YuvHayun thank you

I've fixed the notes you added and opened a new PR:
https://github.com/demisto/content/pull/32765

**Pull request closed without merging**